include/ipr/interface

```diff
@@ -433,6 +433,7 @@ namespace ipr {
    struct Optional {
       Optional(const T* p = nullptr) : ptr { p } { }
       const T& get() const { return *util::check(ptr); }
+      const T* get_ptr() const { return ptr; }
```

Comment:
This change is not correct: no pointer in the interface classes operations.  Why is this operation needed?